src/wesnoth.cpp

```diff
@@ -938,6 +948,15 @@ int main(int argc, char** argv)
 		fprintf(stderr, "Couldn't initialize SDL: %s\n", SDL_GetError());
 		return(1);
 	}
+	
+#ifndef _WIN32
+	struct sigaction terminate_handler;
+	terminate_handler.sa_handler = wesnoth_terminate_handler;
+	terminate_handler.sa_flags = 0;
+	terminate_handler.sa_mask = 0;
```

Comment:
I'm getting a compile error on this line now, this is implemented as a struct in Linux so you cannot simply assign 0 to it.
`sigemptyset(&terminate_handler.sa_mask);` will do the right think on Linux and it looks like it's available on OSX too.